apps/emqx/src/emqx_persistent_session_ds/emqx_persistent_session_ds_shared_subs.erl

```diff
@@ -725,51 +720,3 @@ is_stream_fully_acked(_, _, #srs{
     true;
 is_stream_fully_acked(Comm1, Comm2, #srs{last_seqno_qos1 = S1, last_seqno_qos2 = S2}) ->
     (Comm1 >= S1) andalso (Comm2 >= S2).
-
-%%--------------------------------------------------------------------
-%% Formatters
-%%--------------------------------------------------------------------
```

Comment:
I tried to touch on that in the commit message.

> By default, these are _not_ optimized out in release-ish builds, so it's best to put terms into contexts as is.

Besides, in general I'd argue this is perfectly fine to pollute debug logs and traces with such information. It might be needed for the investigation after all, and the pollution sounds like a problem that's best solved in post-processing.

Reply:
The the trace points are under debug 🤔 but even if the log level is set to that in production, pretty-formatting a stream may be even more resource and memory consuming

Reply:
Yes, but the [context will be computed first](https://github.com/kafka4beam/snabbkaffe/blob/b59298334ed349556f63405d1353184c63c66534/include/trace_prod.hrl#L11-L12) anyway (unless I'm missing something), so in the most common case we'll just spend compute resources and then won't log anything.